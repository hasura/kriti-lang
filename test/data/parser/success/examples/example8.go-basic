{
  "author": {
    "name": {{$.event.name}},
    "age": {{$.event.age}},
    "articles": [
{{ range _, x := $.event.author.articles }}
  {{ if x.published }}
      {
        "id": {{x.id}},
        "title": {{x.title}}
      }
  {{ else }}
      null
  {{ end }}
{{ end }}
    ]
  }
}
