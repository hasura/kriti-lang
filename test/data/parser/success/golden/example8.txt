Object
    ( fromList
        [
            ( "author"
            , Object
                ( fromList
                    [
                        ( "articles"
                        , Array
                            [ Range
                                ( SourcePosition
                                    { _sourceName = "sourceName"
                                    , _line = 6
                                    , _column = 0
                                    }
                                , Just
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 16
                                        , _column = 4
                                        }
                                    )
                                ) Nothing "x"
                                [
                                    (
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 6
                                            , _column = 17
                                            }
                                        , Just
                                            ( SourcePosition
                                                { _sourceName = "sourceName"
                                                , _line = 6
                                                , _column = 18
                                                }
                                            )
                                        )
                                    , Obj "$"
                                    )
                                ,
                                    (
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 6
                                            , _column = 18
                                            }
                                        , Just
                                            ( SourcePosition
                                                { _sourceName = "sourceName"
                                                , _line = 6
                                                , _column = 23
                                                }
                                            )
                                        )
                                    , Obj "event"
                                    )
                                ,
                                    (
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 6
                                            , _column = 24
                                            }
                                        , Just
                                            ( SourcePosition
                                                { _sourceName = "sourceName"
                                                , _line = 6
                                                , _column = 30
                                                }
                                            )
                                        )
                                    , Obj "author"
                                    )
                                ,
                                    (
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 6
                                            , _column = 31
                                            }
                                        , Just
                                            ( SourcePosition
                                                { _sourceName = "sourceName"
                                                , _line = 6
                                                , _column = 39
                                                }
                                            )
                                        )
                                    , Obj "articles"
                                    )
                                ]
                                ( Iff
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 7
                                        , _column = 2
                                        }
                                    , Just
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 15
                                            , _column = 0
                                            }
                                        )
                                    )
                                    ( Path
                                        [
                                            (
                                                ( SourcePosition
                                                    { _sourceName = "sourceName"
                                                    , _line = 7
                                                    , _column = 8
                                                    }
                                                , Just
                                                    ( SourcePosition
                                                        { _sourceName = "sourceName"
                                                        , _line = 7
                                                        , _column = 9
                                                        }
                                                    )
                                                )
                                            , Obj "x"
                                            )
                                        ,
                                            (
                                                ( SourcePosition
                                                    { _sourceName = "sourceName"
                                                    , _line = 7
                                                    , _column = 9
                                                    }
                                                , Just
                                                    ( SourcePosition
                                                        { _sourceName = "sourceName"
                                                        , _line = 7
                                                        , _column = 18
                                                        }
                                                    )
                                                )
                                            , Obj "published"
                                            )
                                        ]
                                    )
                                    ( Object
                                        ( fromList
                                            [
                                                ( "title"
                                                , Path
                                                    [
                                                        (
                                                            ( SourcePosition
                                                                { _sourceName = "sourceName"
                                                                , _line = 10
                                                                , _column = 19
                                                                }
                                                            , Just
                                                                ( SourcePosition
                                                                    { _sourceName = "sourceName"
                                                                    , _line = 10
                                                                    , _column = 20
                                                                    }
                                                                )
                                                            )
                                                        , Obj "x"
                                                        )
                                                    ,
                                                        (
                                                            ( SourcePosition
                                                                { _sourceName = "sourceName"
                                                                , _line = 10
                                                                , _column = 20
                                                                }
                                                            , Just
                                                                ( SourcePosition
                                                                    { _sourceName = "sourceName"
                                                                    , _line = 10
                                                                    , _column = 25
                                                                    }
                                                                )
                                                            )
                                                        , Obj "title"
                                                        )
                                                    ]
                                                )
                                            ,
                                                ( "id"
                                                , Path
                                                    [
                                                        (
                                                            ( SourcePosition
                                                                { _sourceName = "sourceName"
                                                                , _line = 9
                                                                , _column = 16
                                                                }
                                                            , Just
                                                                ( SourcePosition
                                                                    { _sourceName = "sourceName"
                                                                    , _line = 9
                                                                    , _column = 17
                                                                    }
                                                                )
                                                            )
                                                        , Obj "x"
                                                        )
                                                    ,
                                                        (
                                                            ( SourcePosition
                                                                { _sourceName = "sourceName"
                                                                , _line = 9
                                                                , _column = 17
                                                                }
                                                            , Just
                                                                ( SourcePosition
                                                                    { _sourceName = "sourceName"
                                                                    , _line = 9
                                                                    , _column = 19
                                                                    }
                                                                )
                                                            )
                                                        , Obj "id"
                                                        )
                                                    ]
                                                )
                                            ]
                                        )
                                    ) Null
                                )
                            ]
                        )
                    ,
                        ( "name"
                        , Path
                            [
                                (
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 3
                                        , _column = 14
                                        }
                                    , Just
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 3
                                            , _column = 15
                                            }
                                        )
                                    )
                                , Obj "$"
                                )
                            ,
                                (
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 3
                                        , _column = 15
                                        }
                                    , Just
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 3
                                            , _column = 20
                                            }
                                        )
                                    )
                                , Obj "event"
                                )
                            ,
                                (
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 3
                                        , _column = 21
                                        }
                                    , Just
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 3
                                            , _column = 25
                                            }
                                        )
                                    )
                                , Obj "name"
                                )
                            ]
                        )
                    ,
                        ( "age"
                        , Path
                            [
                                (
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 4
                                        , _column = 13
                                        }
                                    , Just
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 4
                                            , _column = 14
                                            }
                                        )
                                    )
                                , Obj "$"
                                )
                            ,
                                (
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 4
                                        , _column = 14
                                        }
                                    , Just
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 4
                                            , _column = 19
                                            }
                                        )
                                    )
                                , Obj "event"
                                )
                            ,
                                (
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 4
                                        , _column = 20
                                        }
                                    , Just
                                        ( SourcePosition
                                            { _sourceName = "sourceName"
                                            , _line = 4
                                            , _column = 23
                                            }
                                        )
                                    )
                                , Obj "age"
                                )
                            ]
                        )
                    ]
                )
            )
        ]
    )